webpack/assets/javascripts/react_app/components/ColumnSelector/helpers.js

```diff
@@ -0,0 +1,87 @@
+const getCheckedStateForCategory = (category = { children: [] }) => {
+  // return true if all children are checked
+  // return null if some children are checked
+  // return false if no children are checked
+  const checked = category.children.map(child => child.checkProps?.checked);
+  if (checked.every(Boolean)) return true;
+  if (checked.some(Boolean)) return null;
+  return false;
+};
+
+export const categoriesFromFrontendColumnData = ({
```

Comment:
This builds the data object that is passed to the `ColumnSelector` component (and I think is passed down in more or less the same form to the Patternfly TreeView component.) I didn't change the data shape at all, so it should still work fine with the legacy page as well. It's a bit tricky because the legacy page mounts the component directly, so it can feed props to it directly from the backend, but we don't have that luxury here, so we need to "rebuild" this object.